admin/authentication/middleware.js

```diff
@@ -2,18 +2,22 @@
 
 const logger = require('../services/log.service').getLogger()
 
-function isAuthenticated (role) {
+function isAuthenticated (roles) {
   return function (req, res, next) {
     if (req.isAuthenticated()) {
       let userRole
       if (req.user) {
         userRole = ((req.user).role || {})
       }
-      logger.debug(`checking authorisation on ${req.url} for role:${role} against userRole:${userRole}`)
-      if (role === undefined || (role !== undefined && userRole && role === userRole)) {
+      logger.debug(`checking authorisation on ${req.url} for role:${roles} against userRole:${userRole}`)
+      // declare single role variable if input is not array of roles
+      const role = !Array.isArray(roles) && roles
+      if (role === undefined ||
+        (role !== undefined && userRole && role === userRole) ||
+        (Array.isArray(roles) && roles.some(r => r === userRole))) {
         return next()
       } else {
-        logger.warn(`could not authorise ${role} against userRole:${userRole}, UserName:${req.user.UserName} ID:${req.user.id}`)
+        logger.warn(`could not authorise ${roles} against userRole:${userRole}, UserName:${req.user.UserName} ID:${req.user.id}`)
```

Comment:
how does `roles` render if it's an array?

Reply:
@GuyHarwood changed so that it displays the message correctly regardless of string or array type.